client/app/scripts/superdesk-desks/aggregate.js

```diff
@@ -54,19 +54,80 @@
                }));
         }))
         .then(angular.bind(this, function() {
-            return preferencesService.get(PREFERENCES_KEY)
-                .then(angular.bind(this, function(preference) {
-                    this.groups = preference != null && preference.groups ? preference.groups : [];
-                    this.loading = false;
+            return this.readSettings()
+                .then(angular.bind(this, function(groups) {
+                    if (groups) {
+                        _.each(groups, angular.bind(this, function(item) {
+                            this.groups.push(item);
+                        }));
+                    }
                     setupCards();
+                    this.loading = false;
                 }));
         }));
 
         /**
+         * Read the setting for current selected workspace(desk or custom workspace)
+         * If the current selected workspace is a desk the settings are read from desk
+         * If the current selected workspace is a custom workspace the settings are read from
+         * user preferences
+         * @return {list} - the list of settings
```

Comment:
Doesn't this method return a _promise_ which is resolved with a list? The docstring is a bit misleading here...